modules/raft/src/main/java/org/apache/ignite/raft/jraft/rpc/impl/RaftGroupServiceImpl.java

```diff
@@ -71,16 +79,16 @@
     private final String groupId;
 
     /** */
-    private final RaftClientMessagesFactory factory;
+    private final RaftMessagesFactory factory;
 
     /** */
-    private volatile Peer leader;
+    private volatile PeerId leader;
 
     /** */
-    private volatile List<Peer> peers;
+    private volatile List<PeerId> peers;
 
     /** */
-    private volatile List<Peer> learners;
+    private volatile List<PeerId> learners;
```

Comment:
It would be better to store Peer in the fields instead of PeerId to avoid multiple conversions on calling getters.
The conversion should be done only on request/response sending/receiving.